zap/src/main/java/org/zaproxy/zap/scan/filters/GenericFilterBean.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2019 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.scan.filters;
+
+import java.util.Collection;
+import java.util.LinkedHashSet;
+import org.apache.commons.collections.CollectionUtils;
+
+/**
+ * Generic Filter Bean which can be applied to any Class Object which implements hashCode and equals
```

Comment:
I think @thc202 had previously indicated that the "bean" nomenclature should be replaced?

Reply:
He has indicated that filterBean (Returned by ScanFilter Interface) should be named FilterResult. I am not sure why bean name nomenclature should be replaced also what should we call these classes which holds information ?

Reply:
I thought the point was to not confuse things with EJB, but perhaps I misunderstood.

Reply:
Yes, that was a general comment not just for `FilterBean`.

> I am not sure why bean name nomenclature should be replaced

Because that has a special meaning, it implies more than just storing information (e.g. one would expect this class to be serializable).

> what should we call these classes which holds information ?

The current name without "Bean", although this is not a filter so it would be better with `FilterData`, `FilterValues`, `FilterContents`...

Reply:
Ok. Will do the changes as suggested.